test/mocks/genMocks.ts

```diff
@@ -0,0 +1,155 @@
+import "mocha";
+import {
+    BigNumber,
+    ContractTransaction,
+    ethers,
+    Signer,
+} from "ethers";
+import { Block, Provider } from "@ethersproject/providers";
+import { GenSale } from "../../src/contracts";
+
+export const feb23TsSeconds = 1677173138;
+export const feb24TsSeconds = 1677264314;
+
+export type GenMocks = {
+    signer: Partial<Signer>;
+    genSale: Partial<GenSale>;
+    currentSaleConfiguration: [
+        string,
+        BigNumber,
+        BigNumber,
+        BigNumber,
+        string,
+        BigNumber,
+        BigNumber,
+        BigNumber
+    ] & {
+        sellerWallet: string;
+        parentDomainId: BigNumber;
+        salePrice: BigNumber;
+        amountForSale: BigNumber;
+        mintlistMerkleRoot: string;
+        startingMetadataIndex: BigNumber;
+        folderGroupID: BigNumber;
+        transactionLimit: BigNumber;
+    };
+    currentSaleStarted: boolean;
+    currentAmountForSale: BigNumber;
+    currentTransactionLimit: BigNumber;
+    currentSalePrice: BigNumber;
+    currentDomainsSold: BigNumber;
+    currentSaleStartTimestamp: BigNumber;
+    currentLatestBlock: Partial<Block>;
+    currentSaleId: string;
+    currentSaleCounter: BigNumber;
+    currentPauseStatus: boolean;
+    currentSignerFunds: ethers.BigNumber;
+    currentDomainsPurchasedByAccount: BigNumber;
+    reset: () => void;
+};
+
+export function getDefaultMockGenSaleConfiguration():
+    [
+        string,
+        BigNumber,
+        BigNumber,
+        BigNumber,
+        string,
+        BigNumber,
+        BigNumber,
+        BigNumber
+    ] & {
+        sellerWallet: string;
+        parentDomainId: BigNumber;
+        salePrice: BigNumber;
+        amountForSale: BigNumber;
+        mintlistMerkleRoot: string;
+        startingMetadataIndex: BigNumber;
+        folderGroupID: BigNumber;
+        transactionLimit: BigNumber;
+    } {
+    return {
+        sellerWallet: "0x1234567890123456789012345678901234567890",
+        parentDomainId: BigNumber.from(123),
+        salePrice: ethers.utils.parseEther("1.0"), // 1 ^ 18th
+        mintlistSaleDuration: BigNumber.from(3600), // ~ 1hr (seconds)
+        amountForSale: BigNumber.from(50),
+        mintlistMerkleRoot: "0xabcdef1234567890abcdef1234567890abcdef12",
+        startingMetadataIndex: BigNumber.from(1),
+        folderGroupID: BigNumber.from(2),
+        transactionLimit: BigNumber.from(10),
+    } as any;
```

Comment:
Consider the above comments to try to turn the return type for this function into an interface or its own type. This way you can do something like below which is both cleaner IMO and safe, because `as` doesn't actually enforce any typing it just causes the compiler to be quiet. And avoiding an explicit `any` would be nice too
```
const config: SaleConfig = {
  ...
}
return config;
```

Reply:
Alright, this is now returning an object inheriting from the SaleConfig interface, so no any is needed anymore